src/senaite/core/browser/form/adapters/sampletype.py

```diff
@@ -0,0 +1,52 @@
+# -*- coding: utf-8 -*-
+#
+# This file is part of SENAITE.CORE.
+#
+# SENAITE.CORE is free software: you can redistribute it and/or modify it under
+# the terms of the GNU General Public License as published by the Free Software
+# Foundation, version 2.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+# FOR A PARTICULAR PURPOSE. See the GNU General Public License for more
+# details.
+#
+# You should have received a copy of the GNU General Public License along with
+# this program; if not, write to the Free Software Foundation, Inc., 51
+# Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+#
+# Copyright 2018-2021 by it's authors.
+# Some rights reserved, see README and LICENSE.
+
+from bika.lims import senaiteMessageFactory as _
+from senaite.core.api import measure as mapi
+from senaite.core.browser.form.adapters import EditFormAdapterBase
+
+
+class EditForm(EditFormAdapterBase):
+    """Edit form adapter for Sample Type
+    """
+
+    def initialized(self, data):
+        return self.data
+
+    def modified(self, data):
+        name = data.get("name")
+        value = data.get("value")
+
+        if name == "MinimumVolume":
+            # Check the value is a valid volume or weight
+            self.add_error_field(name, self.validate_quantity(value))
+
+        return self.data
+
+    def validate_quantity(self, value):
+        """Returns empty string if the value is empty or a valid volume/weight.
+        Returns an error message otherwise
+        """
+        message = ""
+        if value:
+            valid = mapi.is_volume(value) or mapi.is_weight(value)
```

Comment:
Hmm, should we really restrict the sample type just to volume and weight?
Maybe there are sample types where a specific length is needed or a certain amount of pieces?

Reply:
Not sure. At the moment, this Minimum Volume is not used anywhere, although it may eventually be used to check against the sample's reported volume/weight and when creating partitions. However, I need to think a bit more about it, don't see the benefit on having this "MinimumVolume" in Sample Type content. 

I rather think it would make more sense to have "MinimumVolume" at either "Analysis Profile" or even better, at "AR Template" instead. This would allow us to validate the consistency among Container's capacity, Template/Profile's minimum volume and Sample's volume on Sample registration or edition.

Reply:
> Also we should rather add a field validator to check the value than relying only on the Form Adapter. Therefore I would suggest to add a field validator which is additionally used by the Form Adapter to validate the value immediately.

:+1:

Reply:
> Maybe there are sample types where a specific length is needed or a certain amount of pieces?

I cannot think of any Sample Type that would require a specific length. And regarding to amount of pieces, I would think more about an "aggregated" sample type in such case... will keep thinking

Reply:
> Not sure. At the moment, this Minimum Volume is not used anywhere, although it may eventually be used to check against the sample's reported volume/weight and when creating partitions. However, I need to think a bit more about it, don't see the benefit on having this "MinimumVolume" in Sample Type content.
> 
> I rather think it would make more sense to have "MinimumVolume" at either "Analysis Profile" or even better, at "AR Template" instead. This would allow us to validate the consistency among Container's capacity, Template/Profile's minimum volume and Sample's volume on Sample registration or edition.

Yes, it does not make sense to me at this point as well.

E.g. when we have a sample type "Water", but sometimes we need 200 mL and sometimes 500 mL we are already limited with this approach.

Therefore, I would rather do it as you said in either profile or template to decide according to this information the analyses that can be done or the partitions that need to be created.